File: 102-binary-tree-level-order-traversal/102-binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if not root:
            return []
        
        q, ans = [root], []
        while q:
            nq, lvl = [], []
            for node in q:
                lvl.append(node.val)
                if node.left:
                    nq.append(node.left)
                if node.right:
                    nq.append(node.right)
            ans.append(lvl)
            q = nq
        return ans

File: 102-binary-tree-level-order-traversal/102-binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        ArrayList<List<Integer>> ans = new ArrayList<>();
        if(root == null)
            return ans;
        
        ArrayList<TreeNode> q = new ArrayList<>();
        q.add(root);
        while(q.size() > 0){
            ArrayList<TreeNode> nq = new ArrayList<>();
            ArrayList<Integer> level = new ArrayList<>();
            
            for(TreeNode curr : q){
                level.add(curr.val);
                
                if(curr.left != null)
                    nq.add(curr.left);
                
                if(curr.right != null)
                    nq.add(curr.right);
            }
            
            q = nq;
            ans.add(level);
        }
        
        return ans;
    }
}